### PR TITLE
Allow users to enable bucket keys to save on KMS costs

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
This change allows users to save money when calling KMS by using the feature S3 Bucket Keys. Take a look at this resource:

https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-key.html

The default is set to false, which means this has no affect on existing users of the module. I have tested this locally.

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
@dynamike or @eeeady could I get a review on this change? I attempted to get the example set up so you could use it with terratest.

**Comment by chrisgilmerproj** [comment]
@dynamike  @eeeady  - Just wanted to check in and see if you folks wanted to help me out with a review.

File: examples/encryption/main.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (dynamike) [inline examples/encryption/main.tf:1]: Can you change this path from `examples/encryption` to `examples/ s3-bucket-key` to make it clear what the example is doing?

**Review by dynamike — CHANGES_REQUESTED** [review]
Thanks for the PR. Overall it looks good, I dropped one suggestion around your example naming. I'll approve and merge once this has been done.

File: examples/encryption/main.tf (hunk 1 above)
Comment (chrisgilmerproj) [reply examples/encryption/main.tf]: Done! Thanks for taking a look.

**Review by dynamike — APPROVED** [review]

**Pull request merged**